src/modules/commander/Commander.cpp

```diff
@@ -2259,7 +2258,7 @@ Commander::run()
 				tune_negative(true);
 			}
 
-			if (manual_control_setpoint_updated || safety_updated) {
+			if (_manual_control_switches_sub.update(&_manual_control_switches) || safety_updated) {
```

Comment:
Only because it hasn't been refactored yet. Yes the scattered data dependencies are horrible, and in many cases can be made functional so that you pass the new data to check only when it updates.